plugin/pkg/admission/resourcequota/controller.go

```diff
@@ -382,6 +426,17 @@ func (e *quotaEvaluator) checkRequest(quotas []api.ResourceQuota, a admission.At
 
 	// if we have limited resources enabled for this resource, always calculate usage
 	inputObject := a.GetObject()
+	var priorityClass *string
+	var objectRequiresQuotaStrictly bool
+	priorityClass, err := evaluator.GetPriorityClass(inputObject)
```

Comment:
scopes are not specific to priority per earlier comments.

we should be able to structure this by just passing the input object into a generic matches function that works for all scopes.